### ISSUE TITLE
requirejs - bundling over require(/path/to/folder)

### ISSUE THREAD

**Issue by dragoscirjan** [aurelia/cli#680]
**I'm submitting a bug report / feature request**

I consider this to be both a bug report and a feature request.

* **Library Version:**
0.27.0

**Please tell us about your environment:**
* **Operating System:**
any

* **Node Version:**
any

* **NPM Version:**
any

* **Browser:**
not aplicable

* **Language:**
not aplicable

**Current behavior:**

Right now, all elements of type `require('./module)` are interpreted as `fs.readFileSync(/path/to/module.js)`, however,  i.e. in case of [grapql](https://github.com/graphql/graphql-js) (also it is not the only package I had this issue with) `require(./module)` can also refer to `require(./module/index)` actually. Please see the nodejs require functionality.

**Expected/desired behavior:**

`require` interpretation for the "require" loader, should be able to test if argument is file or folder, and also correctly parse the `folder/index` file along with any file that can be found in that file (again, please see the graphql example).

* **What is the expected behavior?**

Right now, `require` expects only files.

Please try `au build --type env` having 

```
         {
            "name": "graphql",
            "path": "../node_modules/graphql",
            "main": "index"
          }
```
added to the bundle.

* **What is the motivation / use case for changing the behavior?**

There are dozens of projects having this type of usage of the `require` method, which at this point cannot be handled by the aurelia-cli bundler. Don't think everybody likes jumping back and forth from cli to webpack and back ...

**Comment by dragoscirjan** [comment]
Yes, the behavior would be from **nodejs**, but I refuse to believe it can't actually be emulated within **aurelia bundler**. I was able to bundle the package mentioned above with **webpack**. So there should be a solution here too :(. 

We are working now to migrate from aurelia-cli back to webpack. But I don't want this to be the solution.

**Comment by JeroenVinke** [comment]
It's true that webpack modifies & wraps packages in a way that allows it to bundle packages which use nodejs specific things. Another example of this are packages that use the `global` variable (which doesn't exist in a browser environment) without checking its existence first. They simply rely on the bundler to fix it so runs in the browser.

We'll definitely want the bundler to be able to bundle this package (and others that do the same). Although that will take time. I'd suggest to load packages that do this via a script tag

**Comment by Alexander-Taran** [comment]
@3cp should be working by now, right?

**Comment by 3cp** [comment]
Yes, nodejs module resolution algorithm is fully supported (minus support of `.node` file) now.

`require("./foo")` can mean `"./foo/index.js"`, `"./foo/index.json"`, and can be more complex with `./foo/package.json`'s main field.
